### DISCUSSION TITLE
How does On-Demand Revalidation works? Is there an express server running all the time?

### DISCUSSION THREAD

**Discussion by wine-fall** [vercel/next.js#48566] [Help] [answered]
### Summary

I just read the doc about On-Demand Revalidation.

The doc suggests add some `.js` file under `/api` folder. And once you want to update your static pages, just fetch the interface.

So, If I build a program by next.js, is there an express server running all the time for accepting some request?

I mean, if I deploy the program on some cloud server, it will stores the bundled file and also running an express server?

Please tell me how does it actually works?🙏

**Comment by icyJoseph** [comment, marked as answer]
It depends... 

If you do on your machine you do, `next build` and then `next start`, as you revalidate pages, either by using on-demand or the revalidate prop, you must remember that the Next.js server is running and listening to incoming requests, serving either, static or dynamic content. 

When deployed to Vercel, and serverless environments, there's tasks running, which are awaken as requests come in. 

So if you are hosting it yourself there's a next server running. It is just an `http` server though:

- `next start`: https://github.com/vercel/next.js/blob/canary/packages/next/src/cli/next-start.ts 
- `startServer`: https://github.com/vercel/next.js/blob/canary/packages/next/src/server/lib/start-server.ts

You can see the results of building in `.next`, it has a bunch of files that might interest you.

But yeah, that's pretty much it, when you revalidate, it builds a new page, that it then serves when a new request for it arrives.
